src/components/common/DiscordGuildSetup/components/DCServerCard.tsx

```diff
@@ -64,13 +62,13 @@ const DCServerCard = ({
         description={serverData.owner ? "Owner" : "Admin"}
         image={serverData.img || "/default_discord_icon.png"}
       >
-        {isSelected ? (
+        {!isUsedInCurrentGuild && isSelected ? (
           <Button h={10} onClick={onCancel}>
             Cancel
           </Button>
-        ) : isValidating ? (
+        ) : !isUsedInCurrentGuild && isValidating ? (
           <Button isLoading />
-        ) : serverData.isGuilded ? (
+        ) : !isUsedInCurrentGuild && serverData.isGuilded ? (
```

Comment:
Maybe we could refactor this logic, because right now this line means `!(serverData.isGuilded && serverData.guildId === id) && serverData.isGuilded`, which is a bit strange :D What do you think?

Reply:
Agreed, made some changes in https://github.com/guildxyz/guild.xyz/pull/1239/commits/ee6e118a42fb26163662a3d5a1a52399a944495e I will have to test if functionally it is the same, once the preview is deployed
`selectButton` appears twice, but I think this way we have the cleanest conditions, what do you think?

Reply:
Yup, this looks much cleaner to me! Thanks!